### PR TITLE
Portable Consent Tutorials

### PR DESCRIPTION
**Pull request by fabriguespe**
This PR initiates 3 tutorials that navigate developers through the Portable Consent feature in XMTP. These tutorials are crafted to facilitate the integration of consent management into XMTP scenarios such as inboxes, subscriptions and broadcasting.

1. **How to Add Consent to an Existing XMTP Inbox (JS)**: This tutorial demonstrates how to incorporate user consent into an XMTP app provisionally. It includes steps for initializing XMTP clients with tentative consent handling, testing conversation filters, and provisional allow/deny actions.

2. **Enabling Subscribers with XMTP**: Tutorial on implementing a 'Subscribe' button with user consent. This guide covers state management, refreshing consent lists, and handling consent-driven subscription instances.

3. **How to Broadcast with XMTP**: This tutorial explores how to use broadcast messages while ensuring consent compliance. It covers rate limiting, consent verification, and message management based on consent status.

4. **How to Add Consent to an Existing XMTP Inbox (RN)**: This tutorial demonstrates how to incorporate user consent into an XMTP app provisionally. It includes steps for initializing XMTP clients with tentative consent handling, testing conversation filters, and provisional allow/deny actions.

### Preview Links:

1. [How to Add Consent to an Existing XMTP Inbox JS](https://junk-range-possible-git-portableconsenttutorials-xmtp-labs.vercel.app/docs/tutorials/portable-consent/request-inbox)
3. [Enabling Subscribers with XMTP](https://junk-range-possible-git-portableconsenttutorials-xmtp-labs.vercel.app/docs/tutorials/portable-consent/subscribe)
4. [How to Broadcast with XMTP](https://junk-range-possible-git-portableconsenttutorials-xmtp-labs.vercel.app/docs/tutorials/portable-consent/broadcast)
1. [How to Add Consent to an Existing XMTP Inbox RN](https://junk-range-possible-git-portableconsenttutorials-xmtp-labs.vercel.app/docs/tutorials/portable-consent/request-inbox-rn)

### Example Repositories:

Each tutorial is coupled with an example demo repo.

1. [How to Add Consent to an Existing XMTP Inbox JS](https://github.com/fabriguespe/xmtp-ppp-request-inbox)
2. [Enabling Subscribers with XMTP](https://github.com/fabriguespe/xmtp-ppp-subscribe)
3. [How to Broadcast with XMTP](https://github.com/fabriguespe/xmtp-ppp-broadcast)
1. [How to Add Consent to an Existing XMTP Inbox RN](https://github.com/fabriguespe/xmtp-rn-request-inbox)

### CODE REVIEW

**Comment by fabriguespe** [comment]
- Added `stream` to the consent build section
- Added reference to the tutorials from the consent section
- Removed interactive widgets
- Added alpha status on tutorials

**Review by jhaaaa — APPROVED** [review]
Hi there @fabriguespe - thank you so much for these tutorials - amazing!

I left a couple of comments below.

I made some copyedits and style updates. If you are interested, you can see the changes in this commit: https://github.com/xmtp/xmtp-dot-org/pull/599/commits/1149591c2046242b612492f0b0e68750f2dab9fc

File: docs/tutorials/portable-consent/broadcast.md [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+const xmtp = await Client.create(signer, { env: env });
+// Check if the client can message the provided wallet addresses
+const broadcasts_canMessage = await xmtp.canMessage([walletAddresses]);
+```
+
+_This will return an array of booleans that correspond to the wallet addresses. If the boolean is true, the address can receive messages. If the boolean is false, the address cannot receive messages._
+
+## Check if address has given consent to the sender
+
+The `isAllowed` method checks if the wallet address has given consent to the sender. If the address has not given consent, the sender cannot send messages to the address. To learn more about consent, check out the tutorial
```
Comment (jhaaaa) [inline docs/tutorials/portable-consent/broadcast.md:65]: @fabriguespe I think there is a link to the tutorial missing here?

File: docs/tutorials/portable-consent/subscribe.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+Import the necessary XMTP and Ethereum packages. These libraries enable you to create an XMTP client and interact with the Ethereum blockchain.
+
+```jsx
+import React, { useEffect, useState } from "react";
+import { Client } from "@xmtp/xmtp-js";
+import { ethers } from "ethers";
+```
+
+The receiver of the subscription is you. In this case, we'll make it a random wallet generated by `ethers`, but you could send your own wallet address as a prop. This wallet is later going to
```
Comment (jhaaaa) [inline docs/tutorials/portable-consent/subscribe.md:34]: @fabriguespe I think there are a few words missing at the end of this paragraph? TY!

**Comment by fabriguespe** [comment]
- Thanks @jhaaaa  for the needed edits! 
- Is this feature still in Alpha? 
- Based on @nplasterer's feedback tutorials now have more awareness on how to prevent overwriting consent scenarios showcasing sections of our [user-consent](https://xmtp.org/docs/build/user-consent#synchronize-user-consent-preferences) doc and the code has been updated considering the consent list is always on sync while performing actions.

**Comment by nplasterer** [comment]
I'd say with portable consent in production on Converse this feature is no longer in alpha. Not sure if it should be in Development or Production? But it feels very stable at this point.

**Pull request merged**